azure-functions/src/commands/run.rs

```diff
@@ -4,27 +4,80 @@ use crate::{
     logger,
     registry::Registry,
     rpc::{
-        status_result::Status, streaming_message::Content, FunctionLoadRequest,
-        FunctionLoadResponse, FunctionRpcClient, InvocationRequest, InvocationResponse,
+        client::FunctionRpc, status_result::Status, streaming_message::Content,
+        FunctionLoadRequest, FunctionLoadResponse, InvocationRequest, InvocationResponse,
         StartStream, StatusResult, StreamingMessage, WorkerInitResponse, WorkerStatusRequest,
         WorkerStatusResponse,
     },
 };
 use clap::{App, Arg, ArgMatches, SubCommand};
-use futures::{future::lazy, sink::Sink, sync::mpsc::unbounded, Future, Stream};
-use grpcio::{ChannelBuilder, EnvBuilder, WriteFlags};
+use futures::{future::lazy, sync::mpsc::unbounded, Future, Poll, Stream};
+use http::{
+    uri::{Authority, Parts, Scheme, Uri},
+    Request as HttpRequest,
+};
 use log::error;
 use std::cell::RefCell;
 use std::panic::{catch_unwind, set_hook, AssertUnwindSafe, PanicInfo};
-use std::sync::Arc;
-use std::thread;
+use tower_grpc::Request;
+use tower_hyper::{
+    client::Builder as HttpBuilder,
+    util::{Connector, Destination, HttpConnector},
+    Connect,
+};
+use tower_service::Service;
+use tower_util::MakeService;
 
 const UNKNOWN: &str = "<unknown>";
 
 thread_local!(static FUNCTION_NAME: RefCell<&'static str> = RefCell::new(UNKNOWN));
 
 type Sender = futures::sync::mpsc::UnboundedSender<StreamingMessage>;
 
+// TODO: replace with tower-request-modifier when published
```

Comment:
Where can the progress on this be tracked?

Reply:
Is it reasonable to include this change for 0.9.0?

Reply:
I'll link the comment to https://github.com/tower-rs/tower-http/issues/24.  The Tower folks have been very accommodating already, so I'm not going to hold up 0.9.0 for tower-request-modifier to be published.  This implementation is a scaled down version of the request modifier that only does setting the origin.